onnx_tf/frontend.py

```diff
@@ -189,6 +213,46 @@ def _bin_op(cls, node, onnx_op):
     return helper.make_node(
             onnx_op, node.inputs, [node.name], name=node.name, broadcast=1)
 
+  @classmethod
+  def _pool_op(cls, node, onnx_op):
+    auto_pad = node.attr["padding"].decode("UTF-8")
+    auto_pad = "SAME_UPPER" if auto_pad == "SAME" else auto_pad
+    data_format = node.attr["data_format"].decode("UTF-8")
+    nc_index = list(map(lambda i: data_format.find(i), "NC"))
+    spatial_index = list(filter(lambda i: i not in nc_index, list(range(len(data_format)))))
+    strides = list(map(lambda i: node.attr["strides"][i], spatial_index))
+    kernel_shape = list(map(lambda i: node.attr["ksize"][i], spatial_index))
+    return helper.make_node(
+      onnx_op,
+      [node.inputs[0]],
+      [node.name],
+      auto_pad=auto_pad,
+      kernel_shape=kernel_shape,
+      strides=strides
+    )
+
+  @classmethod
+  def handle_avg_pool(cls, node, consts):
+    return cls._pool_op(node, "AveragePool")
+
+  @classmethod
+  def handle_conv2_d(cls, node, consts):
+    auto_pad = node.attr["padding"].decode("UTF-8")
+    auto_pad = "SAME_UPPER" if auto_pad == "SAME" else auto_pad
+    data_format = node.attr["data_format"].decode("UTF-8")
+    nc_index = list(map(lambda i: data_format.find(i), "NC"))
```

Comment:
The logic here is a bit error-prone (e.g., you shouldn't use i to indicate a character; it's quite difficult for others to read). It seems like nc_index is calculated just to mask off indicies in spatial_index. I would just write:
```
spatial_indices = [i for i in range(len(data_format))
				   if data_format[i] not in ["N", "C"]]
```

Reply:
Great...
Why did I get nc_index to get spatial_index.....weird..